File: frontend/src/types/reservation.ts
import type { Passenger } from './passenger'
import type { Trip } from './trip'

export type Reservation = {
  code: string
  status: 'Confirmada' | 'Cancelada'
  trip: Trip
  seat: string
  passenger: Passenger
  total: number
  createdAt: string
  travelDate: string | null
}

File: backend/OnibusExpress.Api/Models/Reservation.cs
namespace OnibusExpress.Api.Models;

public class Reservation
{
    public int Id { get; set; }

    public string Code { get; set; } = string.Empty;

    public int TripId { get; set; }

    public Trip Trip { get; set; } = null!;

    public string Seat { get; set; } = string.Empty;

    public string PassengerName { get; set; } = string.Empty;

    public string PassengerCpf { get; set; } = string.Empty;

    public string PassengerEmail { get; set; } = string.Empty;

    public ReservationStatus Status { get; set; } = ReservationStatus.Confirmada;

    public DateTime CreatedAt { get; set; }
}
